cloudify_hostpool/tests/rest/test_backend.py

```diff
@@ -160,3 +166,33 @@ def _acquire():
     def test_acquire_all_hosts_dead(self):
         self.assertRaises(exceptions.NoHostAvailableException,
                           self.backend.acquire_host)
+
+    @mock.patch('cloudify_hostpool.rest.backend.scan.scan',
+                _mock_scan_alive)
+    def test_get_keyfile(self):
+        key_mock = 'test_key'
+        host_id = 'test_id'
+        keyfile = 'key.pem'
+        host_with_key = {
+            'host': '127.0.0.1',
+            'port': '22',
+            'auth': {'username': 'user', 'keyfile': keyfile},
+            'host_id': host_id,
+            'alive': True,
+            'reserved': False
+        }
+        with open(keyfile, 'w') as f:
+            f.write(key_mock)
+        new_db = tempfile.NamedTemporaryFile()
+        old_backend = self.backend
+        try:
+            self.backend = RestBackend(pool={'hosts': [host_with_key, ]},
+                                       db_file_name=new_db.name)
+            host = self.backend.acquire_host()
+            self.assertEqual(host['auth']['keyfile'], key_mock)
+            host = self.backend.get_host(host['host_id'])
+            self.assertEqual(host['auth']['keyfile'], key_mock)
+        finally:
+            if os.path.exists(keyfile):
+                os.unlink(keyfile)
+            self.backend = old_backend
```

Comment:
also, you dont have to initialize a new instance. if you want to ensure you acquire a specific host. just use backend.storage to manually acquire all other hosts.